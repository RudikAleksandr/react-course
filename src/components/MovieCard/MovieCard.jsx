import React from 'react';
import { Row, Col, Image, Dropdown } from 'react-bootstrap';
import cn from './MovieCard.module.css';
import noImageAvailablePath from '../../assets/images/noImageAvailable.jpeg';


const handleErrorLoadPicture = ({ currentTarget }) => {
  currentTarget.src = noImageAvailablePath;
}

const MovieCard = ({ movie, onSelectOption, onSelectMovie }) => {
  return (
    <Row className={`flex-column ${cn.movieCard}`}>
      <Dropdown onSelect={onSelectOption}>
        <Dropdown.Toggle className={cn.dropdownToggle} variant="" id="dropdown-basic" />
        <Dropdown.Menu align="right" className={cn.dropdownMenu}>
          <Dropdown.Item eventKey="EDIT" className={cn.dropdownItem}>Edit</Dropdown.Item>
          <Dropdown.Item eventKey="DELETE" className={cn.dropdownItem}>Delete</Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
      <Image
        className={cn.imageMovie}
        src={movie.poster_path || noImageAvailablePath}
        alt="Movie screensaver"
        onError={handleErrorLoadPicture}
        onClick={onSelectMovie}
      />
      <Col className={`${cn.colTitleRelease} pl-0 pr-0 mt-3`}>
        <span title={movie.title} className={cn.titleMovie}>{movie.title}</span>
        <span className={cn.releaseDate}>{new Date(movie.release_date).getFullYear()}</span>
      </Col>
      <span className={cn.genresMovie}>{movie.genres.join(', ')}</span>
    </Row>
  )
};

export default MovieCard;
