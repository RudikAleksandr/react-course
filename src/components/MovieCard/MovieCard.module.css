
.imageMovie {
  cursor: pointer;
  max-width: 100%;
  min-height: 82%;
  height: auto;
}

.titleMovie {
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 1.2rem;
  color: #bbb;
  max-width: 80%;
}

.releaseDate {
  padding: 2px 15px;
  border: 1px solid #bbb;
  border-radius: 5px;
  color: #bbb;
}

.genresMovie {
  color: #999;
}

.movieCard:hover .dropdownToggle {
  display: block;
}

.movieCard .dropdownToggle {
  display: none;
  position: absolute;
  right: 10px;
  top: 10px;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #2a202d;
  border: none;
}

.movieCard .dropdownToggle::after {
  color: white;
  border: none;
  font-weight: bold;
  font-size: 1.5rem;
  line-height: 1.5rem;
  content: "â‹®";
}


.movieCard .dropdownMenu {
  background-color: #232323;
  border-radius: 6px;
}

.movieCard .dropdownItem {
  color: white;
}

.movieCard .dropdownItem:hover {
  color: white;
  background-color: #f65261;
}

.colTitleRelease {
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  max-height: 32px;
}