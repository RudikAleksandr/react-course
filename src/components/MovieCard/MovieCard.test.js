
import { render, } from '@testing-library/react';
import MovieCard from './MovieCard';
import { MemoryRouter, Route } from 'react-router';

const getComponent = (props) => (
  <MemoryRouter>
    <Route>
      <MovieCard {...props} />
    </Route>
  </MemoryRouter>
);

describe('MovieCard component', () => {
  test('render MovieCard snapshot', () => {
    const movie = {
      id: 'id',
      poster_path: '',
      title: 'title',
      release_date: new Date(2017, 1, 14),
      genres: []
    };
    const component = getComponent({ movie, onSelectOption: jest.fn() });
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });
});
