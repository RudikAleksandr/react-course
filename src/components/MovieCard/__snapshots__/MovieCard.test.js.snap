// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component render MovieCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-column movieCard row"
  >
    <div
      class="dropdown"
      data-testid="selectOption"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="dropdownToggle dropdown-toggle btn"
        id="dropdown-basic"
        type="button"
      />
    </div>
    <a
      href="/film/id"
    >
      <img
        alt="Movie screensaver"
        class="imageMovie"
        src="noImageAvailable.jpeg"
      />
    </a>
    <div
      class="colTitleRelease pl-0 pr-0 mt-3 col"
    >
      <span
        class="titleMovie"
        title="title"
      >
        title
      </span>
      <span
        class="releaseDate"
      >
        2017
      </span>
    </div>
    <span
      class="genresMovie"
    />
  </div>
</DocumentFragment>
`;
